src/actions.h

```diff
@@ -34,12 +34,54 @@ class Action : public Event
 		explicit Action(LuaScriptInterface* interface);
 
 		bool configureEvent(const pugi::xml_node& node) override;
-		bool loadFunction(const pugi::xml_attribute& attr) override;
+		bool loadFunction(const pugi::xml_attribute& attr, bool isScripted) override;
 
 		//scripting
 		virtual bool executeUse(Player* player, Item* item, const Position& fromPosition,
 			Thing* target, const Position& toPosition, bool isHotkey);
 
+		bool getAllowFarUse() const {
+			return allowFarUse;
+		}
+		void setAllowFarUse(bool v) {
+			allowFarUse = v;
+		}
+
+		bool getCheckLineOfSight() const {
+			return checkLineOfSight;
+		}
+		void setCheckLineOfSight(bool v) {
+			checkLineOfSight = v;
+		}
+
+		bool getCheckFloor() const {
+			return checkFloor;
+		}
+		void setCheckFloor(bool v) {
+			checkFloor = v;
+		}
+
+		std::vector<uint16_t> getIdRange() {
+			return ids;
+		}
+		void addId(uint16_t id) {
```

Comment:
Would not it be better, or more organized, to use the same function as the movements?
- addItemId
- addActionId
- addUniqueId
- ...

Reply:
good call, will do that with the next commit, thanks for pointing it out

Reply:
fixed in [24db279](https://github.com/otland/forgottenserver/pull/2558/commits/24db2794032fc0aa78bcddbc70163ea293a9feb8)